### ISSUE TITLE
Multiple inheritance: Cannot determine type of "X" in base class "Y" error

### ISSUE THREAD

**Issue by alexpirine** [python/mypy#2871]
This is probably more of a question than a real issue.

The following code is considered erroneous (tested in Python 3):

```python
class BaseA:
    class Sub:
        pass

class BaseB:
    class Sub:
        pass

class Test(BaseA, BaseB):
    pass
```

mypy responds with:

```
…: error: Cannot determine type of 'Sub' in base class 'BaseA'
…: error: Cannot determine type of 'Sub' in base class 'BaseB'
```

But it doesn't complain if I don't use multiple inheritance:

```python
class BaseA:
    class Sub:
        pass

class BaseB:
    class Sub:
        pass

class Test(BaseA):
    pass
```

**Comment by gvanrossum** [comment]
I think just the error is a bit confusing; in the M.I. case it is trying to
point out that you are inheriting two unrelated class attributes named
'Sub'.

**Comment by JukkaL** [comment]
Based on the error message, it looks likely that the multiple inheritance validity check doesn't expect to see nested classes and gets confused.

**Comment by alexpirine** [comment]
I have the impression that it gets confused, because the following code passes without errors:

```python
class BaseA:
    sub = 1   

class BaseB:
    sub = 1 

class Test(BaseA, BaseB):
    pass
```

I don't really understand what's wrong with nested classes.

**Comment by gvanrossum** [comment]
Well, for one thing, two integers always have the same type. But two independently declared classes (even if they have the same name and contents) are not considered to have the same type.

**Comment by JukkaL** [comment]
Marked as a bug, because the error message is confusing.

**Comment by alexpirine** [comment]
As a side note, I stumbled upon this issue by trying to use two different mixins in Django:

```python
class CreatedDateMixin(models.Model):
    date_created = models.DateTimeField(_("created on"), auto_now_add=True)

    class Meta:
        abstract = True


class UpdatedDateMixin(models.Model):
    date_updated = models.DateTimeField(_("updated on"), auto_now=True)

    class Meta:
        abstract = True


class Foo(CreatedDateMixin, UpdatedDateMixin, models.Model):
    name = models.CharField(_("name"), max_length=255, blank=True)
```

**Comment by ilevkivskyi** [comment]
(Raising priority to high since this is a very common idiom in Django.)

**Comment by ilevkivskyi** [comment]
@mkurnikov you might want to try fixing this.

**Comment by ilevkivskyi** [comment]
Note that mypy still emits an error on the original example, because it is technically unsafe. Note however that @mkurnikov works on a mypy plugin for Django that will suppress this error for Django specific use cases.
